lib/utils/navigation_controller.dart

```diff
@@ -0,0 +1,19 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:injectable/injectable.dart';
+import 'package:kayla_flutter_ic/di/di.dart';
+
+final navigationController = getIt.get<NavigationController>();
+
+@singleton
+class NavigationController with ChangeNotifier {
```

Comment:
@nmint8m Can I know why we create a `ChangeNotifier` class for this controller? I think `Navigator` would work fine without `ChangeNotifier`, or you may want to store the route locally for a purpose 🤔

Reply:
Oh sorry, I will remove this `ChangeNotifier`. Firstly I wanted to adopt [this implementation from Medium](https://medium.com/flutter/learning-flutters-new-navigation-and-routing-system-7c9068155ade) by using the **Navigator 2.0 API**. But it consumes the time that I mentioned [here](https://nimble-co.slack.com/archives/C04HDQ53E74/p1676265517230049). I will try that approach later.